lib/puppet/provider/gnupg_key/gnupg.rb

```diff
@@ -73,7 +79,7 @@ def add_key_from_key_content
     path = create_temporary_file(user_id, resource[:key_content])
     command = "gpg --import #{path}"
     begin
-      output = Puppet::Util::Execution.execute(command, :uid => user_id, :failonfail => true)
+      `#{command}`
```

Comment:
Return code... well, tacking this on the end doesn't still results in no output. Running things manually I get no such problems and the output code is 0. *digs further*
```
command = "gpg --import #{path} && echo $?"
```